### DISCUSSION TITLE
onPinch updated in @use-gesture/react

### DISCUSSION THREAD

**Discussion by marcelobitorical** [pmndrs/use-gesture#483] [Q&A]
How would this code from `react-use-gesture` (9.1.3) translate into `@use-gesture/react` (10.2.11)? 🤔 

I'm updating this package in a project, and the first argument of `offset` is throwing me a `NaN`.

```js
  const bind = useGesture({
    onPinch: ({ offset: [d] }) => {
      // Here goes a manipulation of the d argument
    },
  });
```

I started studying the difference between the new parameters of `onPinch`. The property `offset` is still there. So I don't know why this doesn't work 🤔.

So many thanks in advance! 😄

**Comment by dbismut** [comment]
Can you please share a sandbox?

**Reply by marcelobitorical** [reply]
Hi David! thanks for replying.
I reproduce a similar example on sandbox, and it **kind of** works.

With deprecated package: [https://codesandbox.io/s/condescending-glitter-d3wptq?file=/src/App.js](https://codesandbox.io/s/condescending-glitter-d3wptq?file=/src/App.js)
With the new one: [https://codesandbox.io/s/brave-waterfall-o0gdor?file=/src/App.js:204-246](https://codesandbox.io/s/brave-waterfall-o0gdor?file=/src/App.js:204-246)

With the same code and similar gestures, there are very different values for `offset`. But as I told you, in my case, I didn't receive any values, just `NaN`.

I tested on a tablet.
This is with the deprecated library:
![Screenshot_2022-04-13-16-04-08-662](https://user-images.githubusercontent.com/87870514/163270573-78588406-b8f4-4f1c-a8a6-60bd134bed55.jpeg)
This is with the new one:
![Screenshot_2022-04-13-16-05-28-320](https://user-images.githubusercontent.com/87870514/163270585-ea919856-bdad-43e6-8d26-37c7d6413dc2.jpeg)

I just use the `bind` `useGesture` exactly as it is in the code I wrote. With a `{...bind()}` integration into an SVG element. Nothing more.

**Reply by dbismut** [reply]
Hey, in v10 you're getting scale instead of distance. Thus it's more `offset: [s, a]`, with `s` standing for `scale`. This should make 99% of use cases easier to manage. Does that help?

**Reply by marcelobitorical** [reply]
🤔 Thanks! It could help. I will consider it, but, where does the distance go? in which property I can get it?

**Reply by dbismut** [reply]
@marcelobitorical should be in `state.values = [d, a]`. But again, very unlikely you actually need distance.

**Reply by marcelobitorical** [reply]
Hi @dbismut!
Yes. I think I will use `scale`. But anyway, I still can't get any values from my project when I update the library.
I tried installing the [earliest non-beta version](https://www.npmjs.com/package/@use-gesture/react) of this library and it's giving me the same `NaN` values.
This package updating is surprisingly hard taking into account that [according to the documentation](https://use-gesture.netlify.app/docs/faq/#why-use-react-spring-instead-of-reactusestate) **@use-gesture** aims at being platform agnostic; and that the version `9.1.3` doesn't seems too far from `10.2.11`.

**Reply by dbismut** [reply]
There's a strong chance, although I can't be sure, that the fact that you're getting NaN values comes from your implementation. Please share a sandbox. 

Also 9 being close to 10 doesn't mean that upgrading would be easy. 10 is a major new version. And I'm not sure what this lib being platform agnostic has to do with anything here.

**Reply by marcelobitorical** [reply]
> There's a strong chance, although I can't be sure, that the fact that you're getting NaN values comes from your implementation. 

Yes! I agree. Just that I would like to know what things have changed since that version. I'm looking for differences in the source code of `use-gesture` or some _Updating Guide_.

> Also 9 being close to 10 doesn't mean that upgrading would be easy. 10 is a major new version. And I'm not sure what this lib being platform agnostic has to do with anything here.

I updated the dependency because there is a deprecation warning that comes from `react-use-gesture`. As I read in the FAQ this new package brings a [platform-agnostic](https://use-gesture.netlify.app/docs/faq/#what-is-the-difference-with-react-use-gesture) library. But apparently, there are more changes 😢.

Anyway, I'm still working on a solution in my implementation. (I can't share a sandbox, because the project is huge and private). If there's any path that some `use-gesture` users can follow I will share it here! 🙌
